src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/PolymorphicTypeResolver.cs

```diff
@@ -231,6 +231,71 @@ public bool TryGetDerivedJsonTypeInfo(object typeDiscriminator, [NotNullWhen(tru
             return result;
         }
 
+        /// <summary>
+        /// Walks the type hierarchy above the current type for any types that use polymorphic configuration.
+        /// </summary>
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:UnrecognizedReflectionPattern",
+            Justification = "The call to GetInterfaces will cross-reference results with interface types " +
+                            "already declared as derived types of the polymorphic base type.")]
+        internal static JsonTypeInfo? FindNearestPolymorphicBaseType(JsonTypeInfo typeInfo)
+        {
+            Debug.Assert(typeInfo.IsConfigured);
+
+            if (typeInfo.PolymorphismOptions != null)
+            {
+                // Type defines its own polymorphic configuration.
+                return null;
```

Comment:
(nit) Why not return `typeInfo` in this case?

Reply:
Assuming we consider a type to be an ancestor type of itself, we could conceivably return the type itself in that case. Given how that method is called though:

https://github.com/dotnet/runtime/pull/84768/files/0ddca4920da669d9117acc63f38590a2c98f92fa#diff-c0c6a6581f2a19df70177a8be2af804b0c09915cde3ab47841281fdf6aeadf77

it wouldn't be particularly useful to do so. Perhaps I could update the method name to something that better communicates the real semantics?

Reply:
I think what you have is fine. Looking at the caller of this made me realize this was a bad idea.